File: example03adapters/src/main/java/foo/bar/example/foreadapters/ui/playlist/PlaylistsActivity.java
package foo.bar.example.foreadapters.ui.playlist;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentTransaction;

import foo.bar.example.foreadapters.R;


public class PlaylistsActivity extends FragmentActivity {

    public static void start(Context context) {
        Intent intent = build(context);
        context.startActivity(intent);
    }

    public static Intent build(Context context) {
        Intent intent = new Intent(context, PlaylistsActivity.class);
        return intent;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.common_activity_base);

        if (savedInstanceState == null) {
            setFragment(
                    PlaylistsFragment.newInstance(),
                    PlaylistsFragment.class.getSimpleName());
        }
    }

    private void setFragment(Fragment fragment, String fragmentTag) {
        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(
                R.id.content_main,
                fragment,
                fragmentTag);
        fragmentTransaction.commitAllowingStateLoss();
    }

}

File: example-kt-03adapters/src/main/java/foo/bar/example/foreadapterskt/ui/playlist/PlaylistsActivity.kt
package foo.bar.example.foreadapterskt.ui.playlist

import android.os.Bundle
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.LinearLayoutManager
import co.early.fore.core.observer.Observer
import co.early.fore.core.ui.SyncableView
import co.early.fore.kt.core.logging.Logger
import foo.bar.example.foreadapterskt.OG
import foo.bar.example.foreadapterskt.R
import foo.bar.example.foreadapterskt.feature.playlist.diffable.DiffablePlaylistModel
import foo.bar.example.foreadapterskt.feature.playlist.updatable.UpdatablePlaylistModel
import foo.bar.example.foreadapterskt.ui.playlist.diffable.DiffablePlaylistModelAdapter
import foo.bar.example.foreadapterskt.ui.playlist.updatable.UpdatablePlaylistModelAdapter
import foo.bar.example.foreadapterskt.ui.playlist.listdiffer.ListDifferPlaylistAdapter
import kotlinx.android.synthetic.main.activity_playlists.*

class PlaylistsActivity : FragmentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_playlists)
    }
}
